packages/cozy-harvest-lib/src/components/OAuthForm.jsx

```diff
@@ -1,19 +1,81 @@
 import React, { PureComponent } from 'react'
+import PropTypes from 'prop-types'
 
 import Button from 'cozy-ui/react/Button'
 import { translate } from 'cozy-ui/react/I18n'
 
+import OAuthWindow from './OAuthWindow'
+
+const IDLE = 'idle'
+const WAITING = 'waiting'
+
+/**
+ * The OAuth Form is responsible for displaying a form for OAuth konnectors. It
+ * starts the OAuth process
+ */
 export class OAuthForm extends PureComponent {
+  state = {
+    status: IDLE
+  }
+
+  constructor(props, context) {
+    super(props, context)
+    this.endOAuth = this.endOAuth.bind(this)
+    this.handleAccountId = this.handleAccountId.bind(this)
+    this.startOAuth = this.startOAuth.bind(this)
+  }
+
+  endOAuth() {
+    this.setState({ status: IDLE })
+  }
+
+  handleAccountId(accountId) {
+    const { onSuccess } = this.props
+    if (typeof onSuccess === 'function') onSuccess(accountId)
+  }
+
+  startOAuth() {
+    this.setState({ status: WAITING })
+  }
```

Comment:
startOAuth/endOAuth can be regrouped. Also, I think it is a good convention to have handlers (bound to this) start with `handle`.